.build/docker/ubuntu2004_test.docker

```diff
@@ -146,11 +131,26 @@ RUN /bin/bash -c "export CASS_DRIVER_NO_CYTHON=1 CASS_DRIVER_NO_EXTENSIONS=1 \
     && pip3 freeze --user"
 
 # Initialize the CCM git repo as well as this also can fail to clone
-RUN /bin/bash -c "source ${BUILD_HOME}/env3.6/bin/activate && \
-    ccm create -n 1 -v git:trunk test && ccm remove test && \
+RUN /bin/bash -c "source ${BUILD_HOME}/env3.7/bin/activate && \
     ccm create -n 1 -v git:cassandra-4.1 test && ccm remove test && \
     ccm create -n 1 -v git:cassandra-4.0 test && ccm remove test"
 
+# Initialize ccm versions.  right side of each sequence needs to be updated with new releases.
+#  this can be checked with:
+#  `curl -s https://downloads.apache.org/cassandra/ | grep -oP '(?<=href=\")[0-9]+\.[0-9]+\.[0-9]+(?=)' | sort -V | uniq -w 3`
+RUN bash -c 'source ~/env3.7/bin/activate && \
+    for i in {1..11} ; do echo $i ; ccm create --quiet -n 1 -v binary:4.0.$i test && ccm remove test ; done && \
```

Comment:
I'm not sure if it's worth doing this way, but I think we can get 4.x releases from the `downloads'.  WDYT?

```
m@MBP ~ % curl --silent https://downloads.apache.org/cassandra/ | grep -o '\[DIR\]\"> <a href=".*">' | sed 's/\[DIR\]\"> <a href="//;s/\/">//' | grep -E '^([4-4]|[0-9]{2,})\..*$' | 

The output:

4.0.11
4.1.3
```

```
m@MBP ~ % curl --silent https://downloads.apache.org/cassandra/ | grep -o '\[DIR\]\"> <a href=".*">' | sed 's/\[DIR\]\"> <a href="//;s/\/">//' | grep -E '^([4-4]|[0-9]{2,})\..*$' | awk -F. '{ for(i=1; i<=$NF; i++) system("echo Executing command for version "$0", iteration " i) }'

Executing command for version 4.0.11, iteration 1
Executing command for version 4.0.11, iteration 2
Executing command for version 4.0.11, iteration 3
Executing command for version 4.0.11, iteration 4
Executing command for version 4.0.11, iteration 5
Executing command for version 4.0.11, iteration 6
Executing command for version 4.0.11, iteration 7
Executing command for version 4.0.11, iteration 8
Executing command for version 4.0.11, iteration 9
Executing command for version 4.0.11, iteration 10
Executing command for version 4.0.11, iteration 11
Executing command for version 4.1.3, iteration 1
Executing command for version 4.1.3, iteration 2
Executing command for version 4.1.3, iteration 3
```

Reply:
Can you get it into this line:
```
RUN bash -c 'source ~/env3.7/bin/activate && \
    for i in {1..11} ; do echo $i ; ccm create --quiet -n 1 -v binary:4.0.$i test && ccm remove test ; done && \
    for i in {1..3}  ; do echo $i ; ccm create --quiet -n 1 -v binary:4.1.$i test && ccm remove test ; done'
```

i've failed at every attempt so far…

Reply:
I've managed to improve the latest version :-)

```
mmuzaf@Maxims-MBP ~ % curl --silent https://downloads.apache.org/cassandra/ | grep -o '\[DIR\]\"> <a href=".*">' | sed 's/\[DIR\]\"> <a href="//;s/\/">//' | grep -E '^([4-4]|[0-9]{2,})\..*$' | awk -F. '{ for(i=1; i<=$NF; i++) print $1 "." $2 "." i }' 
4.0.1
4.0.2
4.0.3
4.0.4
4.0.5
4.0.6
4.0.7
4.0.8
4.0.9
4.0.10
4.0.11
4.1.1
4.1.2
4.1.3
```

Reply:
Yeah, it seems this is a problem...
What if we create a script`_init_ccm.sh 4.0 4.1` and run it instead?

Reply:
separate ticket for follow up work ?

Reply:
+1 
:-)